framework/base/Model.php

```diff
@@ -781,7 +781,7 @@ public function activeAttributes()
         if (!isset($scenarios[$scenario])) {
             return [];
         }
-        $attributes = $scenarios[$scenario];
+        $attributes = array_values(array_unique($scenarios[$scenario]));
```

Comment:
`array_unique()` will change order of attributes. This may affect validation and create some unexpected results when order of attributes is important. 

Maybe it is better to use `array_keys(array_flip($scenarios[$scenario]))` instead? It should be faster and preserves order of attributes.

Reply:
Indeed. Order is very important.

Reply:
You're right. Not only does `array_unique()` change the order, it is also very slow because of that (it sorts all the remaining items). Guess PHP developers didn't hear about single responsibility. 
On a side note, what do you think about refactoring this line into a method? The intent would be much clearer to read if it said something like `$this->getUniqueScenarioAttributes()`. Maybe even add something to `yii\helpers\ArrayHelper` since php does not have an efficient or easy to understand way to do this operation. Trying to understand what `array_flip(array_flip($array))` does is so much more harder to understand then something like `ArrayHelper::uniqueValues($array)`

Reply:
Personally I'm for adding `ArrayHelper::unique()`, but it was already rejected: #10378